nengo/solvers.py

```diff
@@ -515,39 +513,30 @@ def __init__(self, weights=False, reg=0.1):
         super(NnlsL2, self).__init__(weights)
         self.reg = reg
 
-    def __call__(self, A, Y, rng=None, E=None):
+    def _solve(self, A, Y, rng, E, sigma):
         # form Gram matrix so we can add regularization
-        sigma = self.reg * A.max()
-        G = np.dot(A.T, A)
-        Y = np.dot(A.T, Y)
-        np.fill_diagonal(G, G.diagonal() + sigma)
-        return super(NnlsL2, self).__call__(G, Y, rng=rng, E=E)
+        GA = np.dot(A.T, A)
+        GY = np.dot(A.T, Y)
+        np.fill_diagonal(GA, GA.diagonal() + A.shape[0] * sigma**2)
+        X, info = super(NnlsL2, self).__call__(GA, GY, rng=rng, E=E)
+        # recompute the RMSE in terms of the original matrices
+        info = {'rmses': _rmses(A, Y, X), 'gram_rmses': info['rmses']}
```

Comment:
The original `info` could have other values that users would want. I'd suggest doing `info = {'rmses': _rmses(A, Y, X), 'gram_info': info}, just so we keep everything around.